specs/Standards/MultiToken/Metadata.md

```diff
@@ -0,0 +1,141 @@
+- NEAR's [Non-Fungible Token Metadata Standard](../NonFungibleToken/Metadata.md)
+- Discussion about NEAR's complete NFT standard: #171
+- Discussion about NEAR's complete Multi Token standard: #246
+
+## Interface
+
+Metadata applies at both the class level (`MTBaseTokenMetadata`) and the specific instance level (`MTTokenMetadata`). The relevant metadata for each:
+
+```ts
+
+type MTContractMetadata = {
+  spec: string, // required, essentially a version like "mt-1.0.0"
+  name: string, // required Zoink's Digitial Sword Collection
+}
+
+type MTBaseTokenMetadata = {
+  name: string, // required, ex. "Silver Swords" or "Metaverse 3"
+  id: string, // required a unique identifier for the metadata
+  symbol: string|null, // required, ex. "MOCHI"
+  icon: string|null, // Data URL
+  decimals: string|null // number of decimals for the token useful for FT related tokens
+  base_uri: string|null, // Centralized gateway known to have reliable access to decentralized storage assets referenced by `reference` or `media` URLs
+  reference: string|null, // URL to a JSON file with more info
+  copies: number|null, // number of copies of this set of metadata in existence when token was minted.
+  reference_hash: string|null, // Base64-encoded sha256 hash of JSON from reference field. Required if `reference` is included.
+}
+
+type MTTokenMetadata = {
+  title: string|null, // ex. "Arch Nemesis: Mail Carrier" or "Parcel #5055"
+  description: string|null, // free-form description
+  media: string|null, // URL to associated media, preferably to decentralized, content-addressed storage
+  media_hash: string|null, // Base64-encoded sha256 hash of content referenced by the `media` field. Required if `media` is included.
+  issued_at: string|null, // When token was issued or minted, Unix epoch in milliseconds
+  expires_at: string|null, // When token expires, Unix epoch in milliseconds
+  starts_at: string|null, // When token starts being valid, Unix epoch in milliseconds
+  updated_at: string|null, // When token was last updated, Unix epoch in milliseconds
+  extra: string|null, // Anything extra the MT wants to store on-chain. Can be stringified JSON.
+  reference: string|null, // URL to an off-chain JSON file with more info.
+  reference_hash: string|null // Base64-encoded sha256 hash of JSON from reference field. Required if `reference` is included.
+}
+
+type MTTokenMetadataAll = {
+  base: MTBaseTokenMetadata
+  token: MTTokenMetadata
+}
+```
+
+A new set of functions MUST be supported on the MT contract:
+
+```ts
+// Returns the top-level contract level metadtata
+function mt_metadata_contract(): MTContractMetadata {}
+function mt_metadata_token_all(token_ids: string[]): MTTokenMetadataAll[]
+function mt_metadata_token_by_token_id(token_ids: string[]): MTTokenMetadata[]
+function mt_metadata_base_by_token_id(token_ids: string[]): MTBaseMetadata[]
+function mt_metadata_base_by_metadata_id(base_metadata_ids: string[]): MTBaseTokenMetadata
```

Comment:
What's the difference between `mt_metadata_base_by_metadata_id` and `mt_metadata_base_by_token_id`? I'm slightly confused

Reply:
`mt_metadata_base_by_metadata_id` is allows you to retrieve the MTBaseTokenMetadata by the base_metadata_id. So using the identifier that you might get from mt_tokens you can resolve what base metadata is associated with it. 

`mt_metadata_base_by_token_id` allows you to return the MTBaseTokenMetadata by token_id. So if you have the token id , you can get the base token level metadata as a convenience factor.

They help you craft the experience, so if you want to iterate through a bunch of tokens, but want to limit the amount of data coming back you can save on the data coming back across the wire, by only returning the base token_id so if you have an nft series of alot of entries you can simply make another fetch to resolve the base token metadata for the series.  

Alternatively if you don't care then you can one shot and get all the data for the token id even if its duplicate data.

Reply:
I also fixed the types hopefully this resolves the confusion, if not i'm happy to make this more clear via comment if you think that would be beneficial :)